### ISSUE TITLE
update uploadFile method to accept buffer

### ISSUE THREAD

**Issue by moonway-dev** [google-gemini/generative-ai-js#221]
### Description of the feature request:

Please make uploadFile able to accept buffer or url, uploading from a local machine is very limiting.

### What problem are you trying to solve with this feature?

This will help loading files in Express JS or Spring Boot

**Comment by tanaikech** [comment]
How about this script? This Node.js script demonstrates how to implement resumable uploads using streams with various APIs, such as Gemini, Drive, YouTube, and more. It supports uploading files from both local paths and remote URLs. I originally created this for use with Gemini API and remote file URLs. I hope you find it helpful.
https://github.com/tanaikech/resumableUploadForGoogleAPIs_nodejs

**Comment by luisiacc** [comment]
Even better if you the gemini endpoint can accept a url to upload the file, that way we don't even have to keep the buffer in memory, we can just upload something like an s3 presigned url.

**Comment by thorwebdev** [comment]
Agreed, imagine a environment where the isn't any `fs`, I'd just like to pass in a publicly accessible URI!

**Comment by hkt74** [comment]
This should be fixed by #365
